file-utils/reader/git_https.go

```diff
@@ -0,0 +1,138 @@
+package reader
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"golang.org/x/term"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+const (
+	GitHttpsPrefix = "git_https"
+)
+
+var regexMap = make(map[string]string)
+
+func ReadGitHttps(gitPath string) (InputValue, error) {
+
+	regex := regexp.MustCompile("(((?P<Username>[^:]+)(:(?P<Password>[^:]+))?)@)?((?P<PathToClone>[[:ascii:]]*.[[:ascii:]]*)[[:blank:]](?P<PathToGet>[[:ascii:]]*))")
+
+	result := regex.FindStringSubmatch(strings.TrimPrefix(gitPath, GitHttpsPrefix+"://"))
+
+	for i, name := range regex.SubexpNames() {
+		if i != 0 && name != "" {
```

Comment:
Here you should do better checking for data existence in `result` and then map it.
`i <= len(result)`